homeassistant/components/persistent_notification/__init__.py

```diff
@@ -140,6 +140,15 @@ def async_dismiss(hass: HomeAssistant, notification_id: str) -> None:
     )
 
 
+@callback
+@bind_hass
+def async_dismiss_all(hass: HomeAssistant) -> None:
+    """Remove all notifications."""
+    notifications = _async_get_or_create_notifications(hass)
+    for notification_id in list(notifications):
+        async_dismiss(hass, notification_id)
```

Comment:
The general advice with python is to try to avoid designs that call functions in a loop if you want them to be performant.  I don’t think that will be a problem here unless we expect hundreds of notifications

Reply:
I can make this totally independent.  It's not an issue for me.  I'll just replicate (to an extent) the pop logic from dismiss.

Reply:
We can avoid the whole loop by making a copy of the dict, clearing the notifications dict with .clear(), and dispatching the copy as a remove.

Reply:
Done